driver-sync/src/test/functional/com/mongodb/client/unified/EventMatcher.java

```diff
@@ -99,46 +91,38 @@ public void assertConnectionPoolEventsEquality(final String client, final BsonAr
-                assertEquals(context.getMessage("Expected ConnectionCheckOutStartedEvent"), eventType, "connectionCheckOutStartedEvent");
-            } else if (actual.getClass().equals(ConnectionCheckedOutEvent.class)) {
-                assertEquals(context.getMessage("Expected ConnectionCheckedOutEvent"), eventType, "connectionCheckedOutEvent");
-            } else if (actual.getClass().equals(ConnectionCheckOutFailedEvent.class)) {
-                assertEquals(context.getMessage("Expected ConnectionCheckOutFailedEvent"), eventType, "connectionCheckOutFailedEvent");
+            assertEquals(context.getMessage("Expected event type to match"), eventType, getEventType(actual.getClass()));
+
+            if (actual.getClass().equals(ConnectionCheckOutFailedEvent.class)) {
+                BsonDocument expected = expectedEventDocument.getDocument(eventType);
                 ConnectionCheckOutFailedEvent actualEvent = (ConnectionCheckOutFailedEvent) actual;
                 if (expected.containsKey("reason")) {
-                    assertEquals(context.getMessage(""), expected.getString("reason").getValue(),
+                    assertEquals(context.getMessage("Expected reason to match"), expected.getString("reason").getValue(),
                             getReasonString(actualEvent.getReason()));
                 }
-            } else if (actual.getClass().equals(ConnectionCheckedInEvent.class)) {
-                assertEquals(context.getMessage("Expected ConnectionCheckedInEvent"), eventType, "connectionCheckedInEvent");
-            } else if (actual.getClass().equals(ConnectionCreatedEvent.class)) {
-                assertEquals(context.getMessage("Expected ConnectionCreatedEvent"), eventType, "connectionCreatedEvent");
-            } else if (actual.getClass().equals(ConnectionReadyEvent.class)) {
-                assertEquals(context.getMessage("Expected ConnectionReadyEvent"), eventType, "connectionReadyEvent");
             } else if (actual.getClass().equals(ConnectionClosedEvent.class)) {
-                assertEquals(context.getMessage("Expected ConnectionClosedEvent"), eventType, "connectionClosedEvent");
+                BsonDocument expected = expectedEventDocument.getDocument(eventType);
                 ConnectionClosedEvent actualEvent = (ConnectionClosedEvent) actual;
                 if (expected.containsKey("reason")) {
-                    assertEquals(context.getMessage(""), expected.getString("reason").getValue(),
+                    assertEquals(context.getMessage("Expected reason to match"), expected.getString("reason").getValue(),
                             getReasonString(actualEvent.getReason()));
                 }
-            } else {
-                throw new UnsupportedOperationException("Unsupported event type: " + actual.getClass());
             }
             context.pop();
         }
         context.pop();
     }
 
+    private String getEventType(final Class<?> eventClass) {
```

Comment:
Ideally, this method should be static. However, I don't want to further "delay" merging the PR.

Reply:
Done

Reply:
🥳

Reply:
Delaying is not a problem.  Happy to fix it.